### PR TITLE
Kill some DB4 workarounds, improve Open Graph, add Twitter Cards metadata

### PR DESCRIPTION
**Pull request by sknorr**
@tomschr please review. Commit-by-commit is probably best.

### CODE REVIEW

**Comment by sknorr** [comment]
Toms asking about "og:prefix" ... research

**Review by tomschr — COMMENTED** [review]
Here is the question

File: suse2021-ns/xhtml/docbook.xsl [hunk 1]
```diff
@@ -304,54 +342,54 @@
...
-    <meta property="og:description" content="{$meta-og.description}"/>
+    <meta property="twitter:image" content="{$socialmedia.preview}"/>
     <xsl:text>&#10;</xsl:text>
-    <meta property="og:url" content="{$canonical.url}"/>
   </xsl:if>
 
+  <!-- The following properties "work" without a canonical URL being defined
+  (but both the Open Graph and Twitter Cards implementations are incomplete
+  without the above tags, better than nothing though). -->
+  <meta property="og:title" content="{$socialmedia.title}"/>
```
[the remark this answers is not in the document]
Comment (sknorr) [reply suse2021-ns/xhtml/docbook.xsl:360]: Clearly, this is something you can do, but it does not appear to be a practical concern. All of the below implementations work fine together with `og:` meta tags:

1. The [Red Hat docs](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/8.4_release_notes/index) clearly do it and this is what their HTML tags look like:
```
<html lang="en" dir="ltr" prefix="content: http://purl.org/rss/1.0/modules/content/  dc: http://purl.org/dc/terms/  foaf: http://xmlns.com/foaf/0.1/  og: http://ogp.me/ns#  rdfs: http://www.w3.org/2000/01/rdf-schema#  schema: http://schema.org/  sioc: http://rdfs.org/sioc/ns#  sioct: http://rdfs.org/sioc/types#  skos: http://www.w3.org/2004/02/skos/core#  xsd: http://www.w3.org/2001/XMLSchema# ">
```

2. [The Register](https://www.theregister.com/2022/01/20/ubuntu_2104_eol/) does not do it:
```
<html lang="en">
```

3. [Heise.de](https://www.heise.de/news/Netflix-Aktie-kracht-weil-Kundenzustrom-nachlaesst-6333970.html) does not do it:
```
<html
  lang="de"
  data-responsive
    data-consentmanagement
    class="article-content--scroll-behavior"
>
```

[the remark this answers is not in the document]
Comment (sknorr) [reply suse2021-ns/xhtml/docbook.xsl:360]: ... as such, I'd suggest you file this as an enhancement request for later. For now, I'd like to make sure to wrangle working HTML5 out of the stylesheet (see WIP #449) and you can then enhance the new HTML tag with the `prefix` attribute when time comes.

**Comment by sknorr** [comment]
The one thing we still appear to be missing is `<meta name="title" content="[60 chars tops]">`...

**Pull request merged**